packages/ember-data/lib/system/many-array.js

```diff
@@ -230,7 +230,13 @@ export default Ember.Object.extend(Ember.MutableArray, Ember.Evented, {
     var type = get(this, 'type');
     var record;
 
-    Ember.assert("You cannot add '" + type.typeKey + "' records to this polymorphic relationship.", !get(this, 'isPolymorphic'));
+    var klass = store.modelFactoryFor(type);
```

Comment:
It is pretty spooky that what `type` contains is unknown this far into the code. Shouldn't this normalization step happen earlier, like on `init`?